src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1349,9 +1349,9 @@ async Task ConnectAsync(NpgsqlTimeout timeout, CancellationToken cancellationTok
                 {
                     ipAddresses = await Dns.GetHostAddressesAsync(Host,  combinedToken).ConfigureAwait(false);
                 }
-                catch (OperationCanceledException oce) when (
-                    oce.CancellationToken == combinedToken && !cancellationToken.IsCancellationRequested)
+                catch (OperationCanceledException)
```

Comment:
I'm wondering if there's value in bubbling up the specific OperationCanceledException thrown by Dns.GetHostAddressesAsync, rather than a new one generated by ThrowIfCancellationRequested below (I think that was probably the thinking behind the current logic...)... I'm not sure there is.

But to keep the behavior that way, it may be a bit better to just keep the `!cancellationToken.IsCancellationRequested` in the `when` for the timeout (or am I misunderstanding the issue)?

In any case, I'm also OK with the change as-is and will approve.

Reply:
OK. Am on vacation and don't have time to review more deeply, and in any case the change looks OK, just a bit odd. No problem for merging.